modules/market/src/main/java/com/opengamma/strata/market/interpolator/SquareLinearCurveInterpolator.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.interpolator;
+
+import java.io.Serializable;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+
+/**
+ * The interpolation is linear on y^2. The interpolator is used for interpolation on variance for options.
+ * All values of y must be positive. 
+ */
+final class SquareLinearCurveInterpolator implements CurveInterpolator, Serializable {
+
+    /**
+     * The serialization version id.
+     */
+    private static final long serialVersionUID = 1L;
+    
+    /** Level below which the value is consider to be 0. 
+     * 
+     */
+    private static final double EPS = 1.0E-10;
+    
+    /**
+     * The interpolator name.
+     */
+    public static final String NAME = "SquareLinear";
+    /**
+     * The interpolator instance.
+     */
+    
+    public static final CurveInterpolator INSTANCE = new SquareLinearCurveInterpolator();
```

Comment:
Normally, I'd put the public static constants first, then the private ones. The existing interpolators don't do that, which is odd, but might as well order the constants here.